peekingduck/commands/core.py

```diff
@@ -0,0 +1,204 @@
+from peekingduck.utils.logger import LoggerSetup
+from peekingduck.viewer import Viewer
+
+logger = logging.getLogger(LOGGER_NAME)  # pylint: disable=invalid-name
+
+
+@click.command()
+@click.option("--custom_folder_name", default="custom_nodes")
+def init(custom_folder_name: str) -> None:
+    """Initializes a PeekingDuck project."""
+    print("Welcome to PeekingDuck!")
+    _create_custom_folder(custom_folder_name)
+    _create_pipeline_config_yml()
+
+
+@click.command()
+@click.option(
+    "--config_path",
+    default=None,
+    type=click.Path(),
+    help=(
+        "List of nodes to run. None assumes pipeline_config.yml at current working directory"
+    ),
+)
+@click.option(
+    "--log_level",
+    default="info",
+    help="""Modify log level {"critical", "error", "warning", "info", "debug"}""",
+)
+@click.option(
+    "--node_config",
+    default="None",
+    help="""Modify node configs by wrapping desired configs in a JSON string.\n
+        Example: --node_config '{"node_name": {"param_1": var_1}}'""",
+)
+@click.option(
+    "--num_iter",
+    default=None,
+    type=int,
+    help="Stop pipeline after running this number of iterations",
+)
+@click.option(
+    "--viewer",
+    default=False,
+    is_flag=True,
+    help="Launch PeekingDuck viewer",
+)
+def run(  # pylint: disable=too-many-arguments
+    config_path: str,
+    log_level: str,
+    node_config: str,
+    num_iter: int,
+    viewer: bool,
+    nodes_parent_dir: str = "src",
+) -> None:
+    """Runs PeekingDuck"""
+    LoggerSetup.set_log_level(log_level)
+
+    if config_path is None:
+        curr_dir = _get_cwd()
+        if (curr_dir / "pipeline_config.yml").is_file():
+            config_path = curr_dir / "pipeline_config.yml"
+        elif (curr_dir / "run_config.yml").is_file():
+            deprecate(
+                "using 'run_config.yml' as the default pipeline configuration "
+                "file is deprecated and will be removed in the future. Please "
+                "use 'pipeline_config.yml' instead.",
+                2,
+            )
+            config_path = curr_dir / "run_config.yml"
+        else:
+            config_path = curr_dir / "pipeline_config.yml"
+    pipeline_config_path = Path(config_path)
+
+    if viewer:
+        logger.info("Launching PeekingDuck Viewer")
+        pkd_viewer = Viewer(
+            pipeline_path=pipeline_config_path,
+            config_updates_cli=node_config,
+            custom_nodes_parent_subdir=nodes_parent_dir,
+            num_iter=num_iter,
+        )
+        pkd_viewer.run()
+    else:
+        start_time = perf_counter()
+        runner = Runner(
+            pipeline_path=pipeline_config_path,
+            config_updates_cli=node_config,
+            custom_nodes_parent_subdir=nodes_parent_dir,
+            num_iter=num_iter,
+        )
+        end_time = perf_counter()
+        logger.debug(f"Startup time = {end_time - start_time:.2f} sec")
+        runner.run()
```

Comment:
I missed this when coding up PKD Viewer: think we should also add `start_time=...`, `end_time=...`, `logger.debug(f"Startup time=...` to the `if viewer:` portion above, for timing debug info and completeness.

Reply:
Added